### ISSUE TITLE
stable package fails to launch on ArchLinux - CI build is failing

### ISSUE THREAD

**Issue by pyamsoft** [LizardByte/pacman-repo#19]
### Describe the Bug

Sunshine version 0.23.1-2 from the LizardByte stable repository.

ArchLinux is currently packaging `boost-libs` `1.86` which provides dynamic libraries from `boost_locale` and other sunshine requirements. The current version of the `sunshine` package published by the stable repository is built against `boost-libs 1.83`

The README claims this repository is automatically rebuilt once a day, but it appears that the current CI action that builds the repository is broken, as the `Build Pacman Repo` action has failed for the past couple of days now.

I am leaving this issue just as a notification to the maintainer that the CI is currently broken and the current package will not run on an ArchLinux system without keeping `boost-libs` on the no-longer-ArchLinux-supported version of 1.83.

Thank you!

### Expected Behavior

Sunshine CI should be corrected which will allow this package to be built against the currently supported ArchLinux version of `boost-libs`

### Additional Context

Thank you for your time and hard work maintaining and building the Sunshine project!

**Comment by ReenigneArcher** [comment]
I've seen the build failures, but I haven't had time to investigate yet.

The PKGBUILD spec isn't specifying any version of boost though, so not necessarily sure what's going on there.

**Comment by pyamsoft** [comment]
Hi

Correct me if I'm wrong, but I believe the boost failure is because the new version on ArchLinux is 1.86 which is an soname bump.

Since the sunshine package was last successfully built and deployed when boost was 1.83, thats the name it "remembers", so that's what it looks for.

I believe that once the CI is fixed and a new build is published, since the new build will be built against 1.86 it will fix this problem?

**Comment by ReenigneArcher** [comment]
Correct, the last stable build that succeeded used 1.83. The builds that attempt to use 1.86 are failing.

So the stable build won't work because your local system has a different version of boost, and thus the binary is linked to a different version.

The pre-release/beta doesn't have this problem because we stopped using the distro's version of boost, specifically to try to avoid this problem.

**Comment by pyamsoft** [comment]
Gotcha, sounds like you all have things handled then. I will switch to the beta repository for the time being until the CI for stable is working again.

Feel free to close this issue as I believe discussion has reached a good conclusion - but you may also elect to keep it open in case you want to send future questioners here regarding the ArchLinux stable repo.

Thanks for your good work!

**Comment by ReenigneArcher** [comment]
> sounds like you all have things handled then

A PR could be submitted to this repo to fix the PKGBUILD for stable if desired.

If not, I can close this when we get another stable release published which will hopefully be soonish.

**Comment by pyamsoft** [comment]
I have tried, but I lack the C++ knowledge here to truly be effective.

When using the PKGBUILD from the repository and trying against the latest boost, the CPP build fails with a bunch of boost-related errors. From looking at the CMake, I _think_ Sunshine is expecting to build against boost 1.85? Something must now be different in 1.86 in the boost headers that prevents the build from happening. 

Though I understand this is now more work for you - it seems like the best course of action is to wait for the Sunshine project to have a new stable release, at which point the updated tooling and build process should be enough to close out this issue.

**Comment by ReenigneArcher** [comment]
> I think Sunshine is expecting to build against boost 1.85

I believe you're looking at the master branch, which would be the beta/pre-release version. Here is v0.23.1. https://github.com/LizardByte/Sunshine/tree/v0.23.1

This may help: https://aur.archlinux.org/cgit/aur.git/commit/?h=sunshine&id=86f68ef7a31a23844a3bfdd809636c310ed4d2d1

**Comment by zoobporsor** [comment]
yeah the main version i cant run or install anymore.  the sunshine-git beta works though so ive been using that in the mean time
